File: backend/entities/ProjectStatusHistory.cs
using System;

namespace Pims.Dal.Entities
{
    /// <summary>
    /// ProjectStatusHistory class, provides an entity for the datamodel to manage project status transition history.
    /// </summary>
    public class ProjectStatusHistory : BaseEntity
    {
        #region Properties
        /// <summary>
        /// get/set - The primary key provides a unique identity for the project history.
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// get/set - Foreign key to the project.
        /// </summary>
        public int ProjectId { get; set; }

        /// <summary>
        /// get/set - The project this history is for.
        /// </summary>
        public Project Project { get; set; }

        /// <summary>
        /// get/set - Foreign key to the project workflow.
        /// </summary>
        public int WorkflowId { get; set; }

        /// <summary>
        /// get/set - The project workflow at this point in time.
        /// </summary>
        public Workflow Workflow { get; set; }

        /// <summary>
        /// get/set - Foreign key to the project status.
        /// </summary>
        public int StatusId { get; set; }

        /// <summary>
        /// get/set - The project status at this point in time.
        /// </summary>
        public ProjectStatus Status { get; set; }
        #endregion

        #region Constructors
        /// <summary>
        /// Create a new instance of a ProjectStatusHistory class.
        /// </summary>
        public ProjectStatusHistory() { }

        /// <summary>
        /// Create a new instance of a ProjectStatusHistory class, initializes with specified arguments
        /// </summary>
        /// <param name="project"></param>
        public ProjectStatusHistory(Project project)
        {
            this.ProjectId = project?.Id ?? throw new ArgumentNullException(nameof(project));
            this.Project = project;
            this.WorkflowId = project.WorkflowId ?? throw new ArgumentException($"Parameter 'project.WorkflowId' is required.");
            this.Workflow = project.Workflow;
            this.StatusId = project.StatusId;
            this.Status = project.Status;
        }
        #endregion
    }
}

File: express-api/src/typeorm/Entities/ProjectStatusHistory.ts
import { Entity, PrimaryGeneratedColumn, Index, ManyToOne, JoinColumn, Column } from 'typeorm';
import { ProjectStatus } from '@/typeorm/Entities/ProjectStatus';
import { Project } from '@/typeorm/Entities/Project';
import { SoftDeleteEntity } from './abstractEntities/SoftDeleteEntity';

@Entity()
export class ProjectStatusHistory extends SoftDeleteEntity {
  @PrimaryGeneratedColumn()
  Id: number;

  // Project Relation
  @Column({ name: 'project_id', type: 'int' })
  ProjectId: number;

  @ManyToOne(() => Project, (Project) => Project.Id)
  @JoinColumn({ name: 'project_id' })
  @Index()
  Project: Project;

  // Status Relation
  @Column({ name: 'status_id', type: 'int' })
  StatusId: number;

  @ManyToOne(() => ProjectStatus, (ProjectStatus) => ProjectStatus.Id)
  @JoinColumn({ name: 'status_id' })
  @Index()
  Status: ProjectStatus;
}
